testing/src/test/java/io/stargate/it/BaseOsgiIntegrationTest.java

```diff
@@ -384,26 +447,63 @@ public void baseSetup() throws BundleException, InterruptedException, IOExceptio
         }
 
         System.setProperty("stargate.40_seed_port_override", "7000");
-        stargateHost = dockerHostIP;
+        if (numberOfStargateNodes > 1) {
+          throw new UnsupportedOperationException(
+              String.format(
+                  "Cassandra > 4.0 with %s set to true AND more that one stargate node is not supported ",
+                  SKIP_HOST_NETWORKING_FLAG));
+        }
+        stargateHosts = new ArrayList<>();
+        stargateHosts.add(dockerHostIP);
         seedHost = "127.0.0.1";
         seedPort = 7001;
       }
-
-      // Start stargate and get the persistance object
-      stargateStarter =
-          new Starter(
-              "Test Cluster",
-              version,
-              stargateHost,
-              seedHost,
-              seedPort,
-              datacenter,
-              rack,
-              isDse,
-              !isDse,
-              9043);
-
-      stargateStarter.start();
+      logger.info("Starting: {} stargate nodes", numberOfStargateNodes);
+      for (int i = 0; i < numberOfStargateNodes; i++) {
+        // Start stargate and get the persistence object
+        try {
+          startStargateInstance(seedHost, seedPort, i);
+        } catch (Exception ex) {
+          logger.error(
+              "Exception when starting stargate node nr: " + i + " it will be retried once.", ex);
+          try {
+            startStargateInstance(seedHost, seedPort, i);
+          } catch (Exception ex2) {
+            logger.error("Exception when retrying start of the stargate node nr: " + i, ex2);
+          }
+        } finally {
+          logger.error("Successful starting stargate node nr: {}", i);
+        }
+      }
     }
   }
+
+  private void startStargateInstance(String seedHost, Integer seedPort, int stargateNodeNumber)
+      throws IOException, BundleException {
+    int jmxPort = new ServerSocket(0).getLocalPort();
```

Comment:
Sometimes, for the last stargate instance (the 3rd one in this case), I am getting:
`Port already in use:` even that this port is picked just before the https://github.com/stargate/stargate/pull/93/files#diff-2c65e0c4892da78e91b8b4d3d357e25aR492.
To solve that, I am retrying the start action just once, and it seems that it helps. Let me know if you have a better idea.